File: patterns/expire-binding-based-timer/rules-js/expire-binding-based-timer-reset.js
rules.JSRule({
  name: "Expire-Timer neu starten",
  description: "Aktiviert alle Expire-Timer nach Systemstart",
  triggers: [triggers.SystemStartlevelTrigger(100)],
  execute: (event) => {
    console.log("Expire-Timer werden neu gestartet");
    items.getItem("gResetExpire").members.forEach((timer) => {
      timer.sendCommand(timer.state);
    });
  }
});

File: patterns/expire-binding-based-timer/rules-python/expire_binding_based_timer_reset.py
from openhab import rule, Registry
from openhab.triggers import SystemStartlevelTrigger


@rule(
    name="Expire-Timer neu starten",
    description="Aktiviert alle Expire-Timer nach Systemstart",
    triggers=[SystemStartlevelTrigger(100)],
)
class ExpireTimerNeuStarten:
    def execute(self, module, input):
        self.logger.info("Expire-Timer werden neu gestartet")
        for timer in Registry.getItem("gResetExpire").getAllMembers():
            self.logger.info(timer.getName() + " -> " + str(timer.getState()))
            timer.sendCommand(str(timer.getState()))
